lib/HLSL/HLOperationLower.cpp

```diff
@@ -157,7 +157,7 @@ struct HLObjectOperationLowerHelper {
     // Add invalid first to avoid dead loop.
     HandleMetaMap[Handle] = {DXIL::ResourceClass::Invalid,
                              DXIL::ResourceKind::Invalid,
-                             StructType::get(Type::getVoidTy(HLM.GetCtx()))};
+                             StructType::get(Type::getVoidTy(HLM.GetCtx()), nullptr)};
```

Comment:
This was an interesting one. getVoidTy is marked as a sentinal, which means if you don't provide a null in the last spot, the compiler warns you something bad(tm) could happen. I wasn't aware of that feature before.

Reply:
Curious about why this is.